presto-main/src/main/java/com/facebook/presto/operator/repartition/PartitionedOutputOperator.java

```diff
@@ -385,14 +405,19 @@ public void partitionPage(Page page)
 
         private Page getPartitionFunctionArguments(Page page)
         {
-            Block[] blocks = new Block[partitionChannels.size()];
+            // Fast path for no constants
+            if (partitionConstantBlocks == null) {
```

Comment:
Wonder if you could add the same optimization to `OptimizedPartitionedOutputOperator`?

Reply:
Didn't realize `OptimizedPartitionedOutputOperator` existed, added the same change there.